### PR TITLE
SoapySDR Driver improvements

### PR DESCRIPTION
**Pull request by sjkelly**
Several cleanups, simplifications, and efforts to make the driver more state-less.

Needs:
https://github.com/JuliaComputing/LMS7002M-driver/pull/17

### CODE REVIEW

File: software/soapysdr-xtrx/XTRXDevice.hpp [hunk 1]
```diff
@@ -346,33 +343,8 @@ class DLL_EXPORT SoapyXTRX : public SoapySDR::Device {
...
-        bool underflow;
-
-        bool burst_end;
-        int32_t burst_samps;
-    } ;
-
-    RXStream _rx_stream;
-    TXStream _tx_stream;
+    Stream _rx_stream;
+    Stream _tx_stream;
```
Comment (staticfloat) [inline software/soapysdr-xtrx/XTRXDevice.hpp:347]: Where is `Stream` defined?

File: JuliaProject.toml [hunk 2]
```diff
@@ -10,6 +10,7 @@ Preferences = "21216c6a-2e73-6563-6e65-726566657250"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 SoapyLMS7_jll = "e7ed14a0-13eb-5dc7-93e2-6133b2eb8bed"
 SoapyLoopback_jll = "8d250ce8-a086-56bd-8a36-e4e8f4202680"
+SoapyRTLSDR_jll = "7bed6c1d-f2db-586c-9908-488058499e34"
```
[the remark this answers is not in the document]
Comment (sjkelly) [reply JuliaProject.toml:13]: I have it hooked up to pathfinder.

File: software/soapysdr-xtrx/XTRXDevice.hpp (hunk 1 above)
Comment (sjkelly) [reply software/soapysdr-xtrx/XTRXDevice.hpp:347]: https://github.com/JuliaComputing/xtrx_julia/blob/4f369be139ed9c877d9d96d55d525743eefe466d/software/soapysdr-xtrx/XTRXDevice.hpp#L329-L344

File: JuliaProject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjkelly) [reply JuliaProject.toml:13]: Oh yeah, this shouldn't be on this PR. I branched off Soeren's work... I will clean it up later before it is ready to merge.

**Comment by sjkelly** [comment]
Since I am leaving for holiday, I will leave this one to sit. Though @zsoerenm if you want to test, feel free to merge. The only bug I know of it that I have a bad inverse function for the baseband frequency. I have not debugged this yet, but it should be easy to track down. Otherwise everything should all work as before, but with more features.

**Comment by sjkelly** [comment]
The BB frequency should be fixed in: https://github.com/JuliaComputing/LMS7002M-driver/pull/20

**Pull request merged**